query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -154,14 +155,27 @@ private String convertFilter2String(
           break;
         case CONTAINS_KEY:
           LiteralConstant[] kvp = convertExpressionToMapLiterals(filter.getRhs());
-          builder.append(convertExpressionToMapKeyColumn(filter.getLhs()));
+          builder.append(convertExpressionToMapKeyColumn(getLogicalColumnName(filter.getLhs())));
           builder.append(" = ");
           builder.append(convertLiteralToString(kvp[MAP_KEY_INDEX], paramsBuilder));
           break;
         case CONTAINS_KEYVALUE:
           kvp = convertExpressionToMapLiterals(filter.getRhs());
-          String keyCol = convertExpressionToMapKeyColumn(filter.getLhs());
-          String valCol = convertExpressionToMapValueColumn(filter.getLhs());
+          String logicalColumnName = getLogicalColumnName(filter.getLhs());
+          if (filter.getLhs().getValueCase() == OBJECTIDENTIFIER) {
+            String pathExpression = filter.getLhs().getObjectIdentifier().getPathExpression();
+            kvp[0] =
```

Comment:
Hmm, fits in the existing implementation. I will check once again once we agree on the `ObjectIndetifier ` part of it.